### ISSUE TITLE
Question on Wan2.2 LoRA Training: Same Dataset/Params, But DiffSynth-Studio Captures Better Character Features – Is It Configuration or Project Implementation? Plus High-Noise Model Theme Suitability

### ISSUE THREAD

**Issue by JamesTensor** [kohya-ss/musubi-tuner#565]
Thank you very much for your work; it has helped many people!

I trained Wan2.2 A14B T2V LoRAs using both musubi-tuner and DiffSynth-Studio. I used the same dataset and made every effort to ensure that the training parameters were consistent. However, I found that the LoRA trained with DiffSynth-Studio can better learn the character features from the original data. I generated the following some videos using these LoRA models for performance comparison.

https://github.com/user-attachments/assets/7adc445f-7e1b-4128-8b57-9cc2bfc84049

https://github.com/user-attachments/assets/4cfc11b1-bfcf-4c3b-b02c-9db75c10cc46

https://github.com/user-attachments/assets/47bd40d2-77b8-4b0e-807b-c4e6f252ed1d

https://github.com/user-attachments/assets/8b68d414-e020-4ad0-98d4-b49aaa6c3246

I am a loyal user of musubi-tuner and have been using it for over half a year. Moreover, musubi-tuner has delivered excellent results when training LoRAs for Wan2.1! That's why I want to figure out the reason behind this: are there differences in the implementation between the two projects, or is there an issue with my training configuration in musubi-tuner? 

By comparing the inference videos generated by the low-noise model alone and those by the high-noise model, I found that since the high-noise model of Wan2.2 has been retrained with a large amount of new data, it might not be suitable for the World of Warcraft theme. But DiffSynth-Studio also trained and used the high-noise model, which seems unable to explain the difference in the generated videos between DiffSynth-Studio and musubi-tuner.

# Below are my training and inference scripts for musubi-tuner:

**musubi-tuner Training script:**

1. low noise Dit Training

nohup accelerate launch --num_cpu_threads_per_process 1 --mixed_precision fp16 src/musubi_tuner/wan_train_network.py --task t2v-A14B --dit /root/pretrained_model/wan2.2_t2v_high_noise_14B_fp16.safetensors/split_files/diffusion_models/wan2.2_t2v_high_noise_14B_fp16.safetensors --dataset_config /root/src/musubi_tuner/dataset/moshou/lora_config.toml --mixed_precision fp16 --optimizer_type AdamW8bit --learning_rate 2e-4 --gradient_checkpointing --max_data_loader_n_workers 1 --persistent_data_loader_workers --network_module networks.lora_wan --network_dim 32 --timestep_sampling shift --discrete_flow_shift 12.0 --min_timestep 875 --max_train_epochs 1000 --timestep_boundary 1000 --seed 42 --preserve_distribution_shape --output_dir /root/lora_train/moshou/t2v/high --output_name moshou --fp8_scaled --fp8_base --vae_cache_cpu --save_every_n_steps 200 --flash_attn --split_attn --gradient_accumulation_steps 4 --vae /root/pretrained_model/Wan2.1_VAE.pth --t5 /root/pretrained_model/models_t5_umt5-xxl-enc-bf16.pth --sample_prompts /root/src/musubi_tuner/dataset/moshou/test_prompt.txt --sample_every_n_steps 400 --sample_at_first > /root/log/wan_moshou_t2v_high.txt 2>&1 &

2. high noise Dit Training

nohup accelerate launch --num_cpu_threads_per_process 1 --mixed_precision fp16 src/musubi_tuner/wan_train_network.py --task t2v-A14B --dit /root/pretrained_model/wan2.2_t2v_low_noise_14B_fp16.safetensors/split_files/diffusion_models/wan2.2_t2v_low_noise_14B_fp16.safetensors --dataset_config /root/src/musubi_tuner/dataset/moshou/lora_config.toml --mixed_precision fp16 --optimizer_type AdamW8bit --learning_rate 2e-4 --gradient_checkpointing --max_data_loader_n_workers 1 --persistent_data_loader_workers --network_module networks.lora_wan --network_dim 32 --timestep_sampling shift --discrete_flow_shift 12.0 --min_timestep 0 --max_train_epochs 1000 --timestep_boundary 875 --seed 42 --preserve_distribution_shape --output_dir /root/lora_train/moshou/t2v/low --output_name moshou --fp8_scaled --fp8_base --vae_cache_cpu --save_every_n_steps 200 --flash_attn --split_attn --gradient_accumulation_steps 4 --vae /root/pretrained_model/Wan2.1_VAE.pth --t5 /root/pretrained_model/models_t5_umt5-xxl-enc-bf16.pth --sample_prompts /root/src/musubi_tuner/dataset/huaxue_2/test_prompt.txt --sample_every_n_steps 400 --sample_at_first > /root/log/wan_huaxue_2_t2v_low.txt 2>&1 &

**musubi-tuner Inference script：**
python src/musubi_tuner/wan_generate_video.py --task t2v-A14B --video_size 1280 720 --fp8 --fp8_scaled --fp8_t5 --vae_cache_cpu --video_length 81 --infer_steps 20 --prompt "9527s. Two muscular orcs are having a boxing match in a wooden ring. They have prominent tusks and pointed ears, with their mouths wide open, panting heavily, clenching their fists. One has green skin, and the other has dark skin. The background is a primitive orc gathering place, featuring some hide tents, clan banners, and bonfires." --output_type video --dit /root/pretrained_model/wan2.2_t2v_low_noise_14B_fp16.safetensors/split_files/diffusion_models/wan2.2_t2v_low_noise_14B_fp16.safetensors --dit_high_noise /root/pretrained_model/wan2.2_t2v_high_noise_14B_fp16.safetensors/split_files/diffusion_models/wan2.2_t2v_high_noise_14B_fp16.safetensors --vae /root/pretrained_model/Wan2.1_VAE.pth --t5 /root/pretrained_model/models_t5_umt5-xxl-enc-bf16.pth --attn_mode flash2 --lora_weight /root/lora_train/moshou/t2v/low/moshou-step00002600.safetensors --lora_multiplier 1.0 --lora_weight_high_noise /root/lora_train/moshou/t2v/high/moshou-step00002600.safetensors --lora_multiplier_high_noise 1.0 --save_path /root/generated_video/moshou/t2v/0914 --seed 42

# **Below are my training and inference scripts for DiffSynth-Studio:**

**DiffSynth-Studio Training script:**
1.high noise Dit Training:
accelerate launch examples/wanvideo/model_training/train.py \
  --dataset_base_path /root/dataset/moshou/ \
  --dataset_metadata_path /root/dataset/moshou/train_lora_data.csv \
  --height 240 \
  --width 416 \
  --num_frames 81 \
  --dataset_repeat 3 \
  --model_paths '[
    [
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00001-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00002-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00003-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00004-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00005-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00006-of-00006-bf16.safetensors"
    ],
    "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.1-T2V-1.3B/models_t5_umt5-xxl-enc-bf16.pth",
    "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.1-T2V-1.3B/Wan2.1_VAE.pth"
]' \
  --learning_rate 1e-4 \
  --num_epochs 500 \
  --remove_prefix_in_ckpt "pipe.dit." \
  --output_path "/root/models/train/Wan2.2-T2V-A14B_high_noise_lora" \
  --lora_base_model "dit" \
  --lora_target_modules "q,k,v,o,ffn.0,ffn.2" \
  --lora_rank 32 \
  --max_timestep_boundary 0.417 \
  --min_timestep_boundary 0 \
  --dataset_num_workers 16 \
  --save_steps 100 \
  --use_gradient_checkpointing_offload

2.low noise Dit Training:
accelerate launch examples/wanvideo/model_training/train.py \
  --dataset_base_path /root/dataset/moshou/ \
  --dataset_metadata_path /root/dataset/moshou/train_lora_data.csv \
  --height 240 \
  --width 416 \
  --num_frames 81 \
  --dataset_repeat 3 \
  --model_paths '[
    [
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00001-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00002-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00003-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00004-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00005-of-00006-bf16.safetensors",
        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00006-of-00006-bf16.safetensors"
    ],
    "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.1-T2V-1.3B/models_t5_umt5-xxl-enc-bf16.pth",
    "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.1-T2V-1.3B/Wan2.1_VAE.pth"
]' \
  --learning_rate 1e-4 \
  --num_epochs 500 \
  --remove_prefix_in_ckpt "pipe.dit." \
  --output_path "/root/models/train/Wan2.2-T2V-A14B_low_noise_lora" \
  --lora_base_model "dit" \
  --lora_target_modules "q,k,v,o,ffn.0,ffn.2" \
  --lora_rank 32 \
  --max_timestep_boundary 1 \
  --min_timestep_boundary 0.417 \
  --dataset_num_workers 16 \
  --save_steps 100 \
  --use_gradient_checkpointing_offload

**DiffSynth-Studio Inference script：**
import torch
from diffsynth import save_video
from diffsynth.pipelines.wan_video_new import WanVideoPipeline, ModelConfig

pipe = WanVideoPipeline.from_pretrained(
    torch_dtype=torch.bfloat16,
    device="cuda",
    model_configs=[
        ModelConfig(path=[
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00001-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00002-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00003-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00004-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00005-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/high_noise_model/diffusion_pytorch_model-00006-of-00006-bf16.safetensors",
                    ]),
        ModelConfig(path=[
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00001-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00002-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00003-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00004-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00005-of-00006-bf16.safetensors",
                        "/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.2-T2V-A14B/low_noise_model/diffusion_pytorch_model-00006-of-00006-bf16.safetensors",
                    ]),
        ModelConfig(model_id="Wan-AI/Wan2.2-T2V-A14B", origin_file_pattern="/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.1-T2V-1.3B/models_t5_umt5-xxl-enc-bf16.pth", offload_device="cpu"),
        ModelConfig(model_id="Wan-AI/Wan2.2-T2V-A14B", origin_file_pattern="/root/examples/wanvideo/model_inference/models/Wan-AI/Wan2.1-T2V-1.3B/Wan2.1_VAE.pth", offload_device="cpu"),
    ],
)

pipe.load_lora(
    module=pipe.dit,  
    path="/root/models/train/Wan2.2-T2V-A14B_high_noise_lora/step-2000.safetensors", 
    alpha=1.0  
)

pipe.load_lora(
    module=pipe.dit2,  
    path="/root/models/train/Wan2.2-T2V-A14B_low_noise_lora/step-2000.safetensors",  
    alpha=1.0  
)

pipe.enable_vram_management()

video = pipe(
    prompt="9527s. Two muscular orcs are having a boxing match in a wooden ring. They have prominent tusks and pointed ears, with their mouths wide open, panting heavily, clenching their fists. One has green skin, and the other has dark skin. The background is a primitive orc gathering place, featuring some hide tents, clan banners, and bonfires.",
    negative_prompt="色调艳丽，过曝，静态，细节模糊不清，字幕，风格，作品，画作，画面，静止，整体发灰，最差质量，低质量，JPEG压缩残留，丑陋的，残缺的，多余的手指，画得不好的手部，画得不好的脸部，畸形的，毁容的，形态畸形的肢体，手指融合，静止不动的画面，杂乱的背景，三条腿，背景人很多，倒着走",seed=0, tiled=True,
)
save_video(video, "moshou-test.mp4", fps=15, quality=5)

This comparison is only limited to the World of Warcraft theme and cannot represent the overall performance. In my previous work, I have always considered musubi-tuner to be better than other frameworks, and this is the reason why I have been using musubi-tuner consistently!

Thanks again for the author's great work!

**Comment by kohya-ss** [comment]
In DiffSynth-Studio setting, the timestep range for high-noise LoRA is 0.417 to 1.0, and for low-noise LoRA it is 0.0 to 0.417.
In contrast, in the sd-scripts settings, the timestep range for high-noise LoRA is 857 to 1000 (0.875 to 1.0), and for low-noise LoRA it is 875 to 1000 (0.875 to 1.0).

This is probably the reason. The optimal value for timestep range is unknown, so try adjusting it to the settings in Diffsynth-studio.

**Comment by JamesTensor** [comment]
Thank you for your prompt reply！
I consulted the technical staff of DiffSynth-Studio in the official group, and they said that --min_timestep_boundary 0.417 corresponds to timestep 875, with the only difference being the implementation method within the framework. Therefore, the two are consistent in terms of this parameter.

https://github.com/modelscope/DiffSynth-Studio/blob/main/examples/wanvideo/model_training/lora/Wan2.2-T2V-A14B.sh

**Comment by kohya-ss** [comment]
Hmm, it seems they're using the timestep index instead of the actual timesteps.

It seems they're using 5.0 as the shift value. Try specifying `--discrete_flow_shift 5.0` in Musubi Tuner.

https://github.com/modelscope/DiffSynth-Studio/blob/d93e8738cdbba9d5dddd3b1f75ac8d0b0a12d509/diffsynth/pipelines/wan_video_new.py#L38

**Comment by JamesTensor** [comment]
Thank you so much! After I modified and aligned the discrete_flow_shift configuration, the performance of the tuner project has been significantly improved! Thank you for your contribution!

**Comment by kohya-ss** [comment]
I'm glad to hear that. Please re-open this if any problem persists.
